sentence_transformers/losses/DenoisingAutoEncoderLoss.py

```diff
@@ -114,7 +114,7 @@ def __init__(self, model: SentenceTransformer, decoder_name_or_path: str = None,
                 )
             decoder_base_model_prefix = self.decoder.base_model_prefix
             PreTrainedModel._tie_encoder_decoder_weights(
```

Comment:
Hello!

Well spotted; I see this has been introduced in transformers v4.40.0 [here](https://github.com/huggingface/transformers/pull/29906). Because I'd rather not increment the minimum `transformers` all the way to v4.44.0 yet (I imagine that might cause problems), I propose an alternative solution:

```python
try:
    # Compatibility with transformers <4.40.0
    PreTrainedModel._tie_encoder_decoder_weights(
        model[0].auto_model, self.decoder._modules[decoder_base_model_prefix], self.decoder.base_model_prefix
    )
except TypeError:
    # Compatibility with transformers >=4.40.0
    PreTrainedModel._tie_encoder_decoder_weights(
        model[0].auto_model, self.decoder._modules[decoder_base_model_prefix], self.decoder.base_model_prefix, encoder_name_or_path
    )
```

Reply:
Thanks! @tomaarsen makes sense. It has been updated and now works fine!